### DISCUSSION TITLE
Can hot reload be used while the program is running?

### DISCUSSION THREAD

**Discussion by NMSAzulX** [dotnet/roslyn#71048] [Q&A]
Hello. Can hot reload be used while the program is running?  When I run the program (not f5), the value of MetadataUpdater.isSupported is false. I hope to dynamically generate class A and update the old class A while the program is running. :)

**Comment by huoyaoyuan** [comment]
I don't think it's possible at all. Hot reload will invalidate nearly all cross-type optimizations.

**Comment by tmat** [comment]
@NMSAzulX Yes, Hot Reload works when running a debug build of your application. This is only supported on .NET 6+, not on Desktop Framework.

**Reply by NMSAzulX** [reply]
Thank you. Does Hot Reload work when separated from the code editor?

**Reply by NMSAzulX** [reply]
The working environment of the program is .NET 8.0. But the value of MetadataUpdater.isSupported is false. Not run in code editor. （Double click the .exe.）
